testsuite/tests/apicast/auth/rhsso/openid_rhsso/test_rhsso_wrong_realm.py

```diff
@@ -0,0 +1,68 @@
+"""Tests that accessing APIcast with token from a different RHSSO realm does not work"""
+import pytest
+
+from packaging.version import Version  # noqa # pylint: disable=unused-import
+from testsuite import TESTED_VERSION  # noqa # pylint: disable=unused-import
+
+from testsuite.rhsso import Token
+from testsuite.utils import blame
+
+
+@pytest.fixture(scope="module")
+def user_wrong_realm(rhsso_service_info, request, testconfig):
+    """User in a wrong realm"""
+    realm = rhsso_service_info.rhsso.create_realm(blame(request, "realm2"), accessTokenLifespan=24 * 60 * 60)
+
+    if not testconfig["skip_cleanup"]:
+        request.addfinalizer(realm.delete)
+
+    client = realm.create_client(
+        name=blame(request, "client2"),
+        directAccessGrantsEnabled=True,
+        publicClient=False,
+        protocol="openid-connect",
+        standardFlowEnabled=False)
+
+    username = testconfig["rhsso"]["test_user"]["username"]
+    password = testconfig["rhsso"]["test_user"]["password"]
+    realm.create_user(username, password)
+    return client.oidc_client, username, password
+
+
+@pytest.fixture(scope="function")
+def wrong_realm_token(user_wrong_realm):
+    """Token for a wrong realm"""
+    client, username, password = user_wrong_realm
+    return Token(client.token(username, password))['access_token']
+
+
+@pytest.fixture(scope="function")
+def correct_realm_token(rhsso_service_info, application):
+    """Token for a correct realm"""
+    return rhsso_service_info.access_token(application)
+
+
+@pytest.fixture(scope="module")
+def backends_mapping(custom_backend):
+    """
+    :return: dict in format {path: backend}
+    """
+    return {"/test": custom_backend("backend")}
+
+
+# It doesn't work on 2.13.1
+@pytest.mark.skipif("TESTED_VERSION < Version('2.12')")
```

Comment:
It is not necessary but I think that @pehala can consider to split this test into 2: one for calls which should fail and another one for passing calls.

Reply:
I cannot filter by minor, TESTED_VERSION currently contains only `2.12`

Reply:
Sorry, I've marked wrong line. I've meant that this whole test can be split into passing(20X) asserts and failing(4XX) asserts/http calls.

Reply:
Actually you can and should define patch release @pehala